src/runtime/pkg/containerd-shim-v2/create.go

```diff
@@ -112,20 +112,29 @@ func create(ctx context.Context, s *service, r *taskAPI.CreateTaskRequest) (*con
 		if s.sandbox != nil {
 			return nil, fmt.Errorf("cannot create another sandbox in sandbox: %s", s.sandbox.ID())
 		}
+		s.config = runtimeConfig
+
 		// Here we deal with CDI devices that are cold-plugged (k8s) and
 		// for the single_container (nerdctl, podman, ...) use-case.
 		// We can provide additional directories where to search for
 		// CDI specs if needed. immutable OS's only have specific
 		// directories where applications can write too. For instance /opt/cdi
 		//
 		// _, err = withCDI(ociSpec.Annotations, []string{"/opt/cdi"}, ociSpec)
+		err = resolveCDIAnnotations(ctx, s, ociSpec.Annotations)
```

Comment:
Just for my understanding: We only seem to use resolveCDIAnnotations and WithCDI here. Instead of first adding CDI annotations to the SPEC, then parsing these CDI annotations again (in WithCDI) and removing these annotations later again, can we just not add to the CDI annotations and instead pass devices directly to WithCDI.
Or: Can we just not add to the ociSpec.Annotations, use a separate map, then pass that separate map to WithCDI? This way we'd again not need to remove CDI annotations for the ociSpec.annotations